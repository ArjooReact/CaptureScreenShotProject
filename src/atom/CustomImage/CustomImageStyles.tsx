import { StyleSheet } from "react-native";
import Colors from "../../utility/Colors";
import Dimension from "../../utility/Dimension";
const styles=StyleSheet.create({
    previewImage2: {
        width: Dimension.PreviewImage2Dimen,
        height: Dimension.PreviewImage2Dimen,
        backgroundColor: Colors.WHITE,
        borderRadius: Dimension.ImageBorderRadius,
      },
      previewImage: {
        width: Dimension.PreviewImageDimen, 
        height: Dimension.PreviewImageDimen, 
        backgroundColor: Colors.WHITE
      },
})

export default styles