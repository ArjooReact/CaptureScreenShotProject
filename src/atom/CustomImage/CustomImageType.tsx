export interface CustomImageType{
    uri?:any
}