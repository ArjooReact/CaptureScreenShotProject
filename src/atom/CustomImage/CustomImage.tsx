import React from 'react';
import {StyleSheet, Image} from 'react-native';
import {CustomImageType} from './CustomImageType';
import styles from './CustomImageStyles';

const CustomImage: React.FC<CustomImageType> = ({uri}) => {
  if (uri) {
    return (
      <Image
        source={{uri: uri}}
        style={styles.previewImage}
        resizeMode="contain"></Image>
    );
  } else {
    return (
      <Image
        source={require('../../assets/appimage.png')}
        style={styles.previewImage2}
        resizeMode="contain"
      />
    );
  }
};


export default CustomImage;
