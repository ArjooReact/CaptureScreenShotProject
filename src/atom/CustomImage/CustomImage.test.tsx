import renderer from 'react-test-renderer';
import {test, jest, describe} from '@jest/globals';
import {expect, it, beforeEach} from '@jest/globals';
import CustomImage from './CustomImage';
import { View } from 'react-native';
describe('all mock', () => {
  let mockButtonClick = jest.fn();
  let children=()=>{
    return<View></View>
  }
  test('testcase for matching customImage Snapshot==>', () => {
    const AppRef = renderer.create(
      <CustomImage uri='imagePath' />,
    );
    expect(AppRef.toJSON()).toMatchSnapshot();
  });

  test('testcase for checking presence of dom element for CustomImage==>', () => {
    const AppRef = renderer.create(
        <CustomImage uri='imagePath' />,
    );
    console.log('arzoo test======', AppRef.toJSON());
    let props: any = AppRef.toJSON();
    console.log('Arzoo data...',props)
    expect(AppRef.toJSON()).toBeTruthy();
  });

  test('testcase for checking child element for CustomImage==>', () => {
    const styles = {};
    const AppRef = renderer.create(
        <CustomImage uri='imagePath' />,
    );
    let props: any = AppRef.toJSON();
    expect(props?.type).toBe('Image');
  });
  test('testcase for checking child element for CustomImage==>', () => {
    //const styles={}
    const AppRef: any = renderer.create(
        <CustomImage uri='imagePath' />,
    );
    const styles = AppRef.toJSON()?.props.style;
    console.log('Propsdd.ff....', styles);
    const {width, height, backgroundColor} = styles;
    expect(width).toBe(200);
    expect(height).toBe(200);
    expect(backgroundColor).toBe('#ffffff');
   // expect(borderRadius).toBe(8);
  });
});
