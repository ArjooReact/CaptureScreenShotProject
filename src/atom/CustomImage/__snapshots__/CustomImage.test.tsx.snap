// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`all mock testcase for matching customImage Snapshot==> 1`] = `
<Image
  resizeMode="contain"
  source={
    {
      "uri": "imagePath",
    }
  }
  style={
    {
      "backgroundColor": "#ffffff",
      "height": 200,
      "width": 200,
    }
  }
/>
`;

exports[`all mock testcase for matching customModal Snapshot==> 1`] = `
<Image
  resizeMode="contain"
  source={
    {
      "uri": "imagePath",
    }
  }
  style={
    {
      "backgroundColor": "#ffffff",
      "height": 200,
      "width": 200,
    }
  }
/>
`;
